src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.cs

```diff
@@ -190,6 +192,7 @@ private class TableEntriesFactory : AbstractTableEntriesFactory<DiagnosticData>
                     _projectId = projectId;
                     _documentId = documentId;
                     _id = id;
+                    _errorSource = (id as ErrorSourceId)?.ErrorSource ?? string.Empty;
```

Comment:
Is it possible that id can be *not* an ErrorSourceId for roslyn-produced diagnostics?